include/seastar/coroutine/async_generator.hh

```diff
@@ -0,0 +1,421 @@
+/*
```

Comment:
Why call it async_generator and not generator? Everything's async in seastar.

Reply:
But why rename it?

Reply:
it's renamed. but the net result is that it was replaced. the steps are like:

1. add `async_generator`
2. rename `async_generator` to `generator`.

probably i misread your question. but i am not adding a new generator.

Reply:
ok